src/main/scala/org/allenai/pnp/BsoTrainer.scala

```diff
@@ -46,14 +46,18 @@ class BsoTrainer(val epochs: Int, val beamSize: Int, val maxIters: Int,
     } else {
       example.conditionalExecutionScore
     }
-    val graph = model.getComputationGraph()
 
-    val queue = new BsoPnpQueue[A](beamSize, stateCost, graph, log)
-    val finished = new BsoPnpQueue[A](beamSize, stateCost, graph, log)
+    val queue = new BsoPnpQueue[A](beamSize, stateCost)
+    val finished = new BsoPnpQueue[A](beamSize, stateCost)
     val endContinuation = new PnpEndContinuation[A]()
 
-    val startEnv = env.setLog(log)
-    queue.offer(example.unconditional, env, 0.0, null, null, env)
+    val startEnv = env
+    // TODO(is this right?)
+    val context = PnpInferenceContext.init(model)
+        .addExecutionScore(stateCost)
```

Comment:
Don't add the execution score here. (The way the scores are used here is slightly weird)